addons/sale_purchase/models/purchase_order.py

```diff
@@ -11,11 +11,22 @@ class PurchaseOrder(models.Model):
         "Number of Source Sale",
         compute='_compute_sale_order_count',
         groups='sales_team.group_sale_salesman')
+    has_sale_order = fields.Boolean(
+        "Technical field for whether the purchase order has associated sale orders",
+        compute='_compute_sale_order_count')
 
     @api.depends('order_line.sale_order_id')
     def _compute_sale_order_count(self):
         for purchase in self:
             purchase.sale_order_count = len(purchase._get_sale_orders())
+            purchase.has_sale_order = bool(purchase.sale_order_count)
+
+    @api.depends('order_line.sale_order_id.partner_shipping_id')
+    def _compute_dest_address_id(self):
+        po_with_so = self.filtered('has_sale_order')
+        for order in po_with_so:
+            order.dest_address_id = order._get_sale_orders().partner_shipping_id
+        super(PurchaseOrder, self - po_with_so)._compute_dest_address_id()
```

Comment:
Maybe you can make it look prettier :thinking: 
```suggestion
    def _compute_dest_address_id(self):
        po_with_so_address = set()
        for order in self:
            sale_orders = purchase._get_sale_orders()
            if len(sale_orders.partner_shipping_id) != 1:
                continue
            order.dest_address_id = sale_orders.partner_shipping_id
            po_with_so_address.add(order.id)
        super(PurchaseOrder, self - self.env['purchase.order'].browse(po_with_so_address))._compute_dest_address_id()
```